lib/psql.js

```diff
@@ -18,17 +22,31 @@ function * exec (db, query) {
     PGPORT: db.port,
     PGHOST: db.host
   })
-  let {stdout, error: err, status} = spawnSync('psql', ['--command', query], {env, encoding: 'utf8', stdio: [0, 'pipe', 2]})
-  if (err) {
-    if (err.code !== 'ENOENT') throw err
-    cli.error(`The local psql command could not be located.
-For help installing psql, see https://devcenter.heroku.com/articles/heroku-postgresql#local-setup`)
-    process.exit(1)
-  }
+}
+
+function * exec (db, query) {
+  const stripEOF = require('strip-eof')
+  const {spawnSync} = require('child_process')
+
+  debug(query)
+  let {stdout, error: err, status} = spawnSync('psql', ['--command', query], {env: env(db), encoding: 'utf8', stdio: [0, 'pipe', 2]})
+  handleError(err)
   if (status !== 0) process.exit(status)
   return stripEOF(stdout)
 }
 
+function * interactive (db) {
+  const {spawnSync} = require('child_process')
+  let name = db.addon.config_vars[0].replace(/^HEROKU_POSTGRESQL_/, '').replace(/_URL$/, '')
+  let prompt = `${db.addon.app.name}::${name}%R%# `
+  let {error: err, status} = spawnSync('psql',
+    ['--set', `PROMPT1=${prompt}`, '--set', `PROMPT2=${prompt}`],
+    {env: env(db), stdio: 'inherit'})
```

Comment:
Oh also, do you know the behavior with signals? I think that affects both the interactive and non-interactive cases: e.g., if I have `SELECT pg_sleep(1000000)`, I should be able to kill it in both cases with `Ctrl-C`. I checked the Ruby `pg:psql` and this does work.